### PR TITLE
[W5.4][W15-A4] Huang Chi Ying

### PR DESCRIPTION
**Pull request by john19950730**
Implemented 'pfind' command to find contacts by phone number.
Included regression test cases to test 'pfind' command.
Updated user guide and help command to reflect addition of 'pfind' command.
Additional bug fixes done as well.

### CODE REVIEW

**Comment by john19950730** [comment]
JUnit test cases for FindPhoneCommand added as well.
Considering adding JUnit tests for Parser later.

**Comment by 1moresec** [comment]
Nice approach on finding by part of the phone number. Updating the learning outcomes part makes the diagram clear. Suggest to only import needed class from java util package in FindPhoneCommand instead of importing  the whole package.

File: doc/LearningOutcomes.md [hunk 1]
```diff
@@ -69,10 +69,9 @@ Here's a slightly more difficult exercise.
 
 * [se-edu/se-book: Error Handling: Exceptions](https://se-edu.github.io/se-book/errorHandling/exceptions/)
 
-##### Exercise: Handle 'file deleted' situation 
+##### Exercise: Handle 'file readonly' situation 
 
-* The current code does not handle the situation where the user deletes the storage file while the
-  AddressBook program is running. Use exceptions to handle that situation.
+* The current code does not handle the situation where the user accidentally makes the storage file _read only_ while the AddressBook program is running. Use exceptions to handle that situation better. e.g. instead of crashing, the program can inform the user of the problem.
```
Comment (fanyiii) [inline doc/LearningOutcomes.md]: Good job in updating DevelopGuide as well!

File: src/seedu/addressbook/commands/FindPhoneCommand.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+     * @param inclusiveTerms a list for inclusive search terms to be added to
+     * @param exclusiveTerms a list for exclusive search terms to be added to
+     */
+    private void classifySearchTerms(String[] numbersString, List inclusiveTerms, List exclusiveTerms) {
+        for (String number : numbersString) {
+            int phoneNumberTerm = Integer.parseInt(number);
+            if (phoneNumberTerm < 0)
+                exclusiveTerms.add(Math.abs(phoneNumberTerm));
+            else
+                inclusiveTerms.add(phoneNumberTerm);
```
Comment (fanyiii) [inline src/seedu/addressbook/commands/FindPhoneCommand.java]: According to our [Java coding standard](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#conditionals), single statement conditionals should still be wrapped by curly brackets.

File: src/seedu/addressbook/commands/FindPhoneCommand.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    /**
+     * Checks if specified person has phone number containing any sequence in list of search terms
+     * @param person a ReadOnlyPerson object to check
+     * @param searchTerms a list of search terms to check against
+     * @return true if specified person has phone number that contains any sequence in list of search terms,
+     *          false otherwise
+     */
+    private boolean checkPersonPhoneNumberAgainstSequencesInList(ReadOnlyPerson person, List<Integer> searchTerms) {
+        for (int singleSearchTerm : searchTerms)
```
Comment (fanyiii) [inline src/seedu/addressbook/commands/FindPhoneCommand.java]: Again, curly brackets missing here

File: test/java/seedu/addressbook/commands/FindPhoneCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
+package seedu.addressbook.commands;
+
+import org.junit.Test;
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.util.TypicalPersons;
+
+import java.util.*;
```
Comment (fanyiii) [inline test/java/seedu/addressbook/commands/FindPhoneCommandTest.java]: Import statements should be stated explicitly.

File: test/java/seedu/addressbook/commands/FindPhoneCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        //no match
+        assertFindPhoneCommandBehavior(new String[] {"7", "11119", "22229", "33339"}, Collections.emptyList());
+        /* Multiple Inclusive and Exclusive Search Terms */
+        //multiple match
+        assertFindPhoneCommandBehavior(new String[] {"9", "-393"}, Arrays.asList(td.amy, td.bill));
+        //single match
+        assertFindPhoneCommandBehavior(new String[] {"9", "-33", "-191"}, Arrays.asList(td.bill));
+        //no match
+        assertFindPhoneCommandBehavior(new String[] {"911", "-9"}, Collections.emptyList());
+    }
```
Comment (fanyiii) [inline test/java/seedu/addressbook/commands/FindPhoneCommandTest.java]: Very thorough test cases!

**Pull request closed without merging**